### PR TITLE
support conditional-exports in node 12+

### PR DESCRIPTION
**Pull request by robbtraister**
This is the code change described here: https://github.com/mesqueeb/merge-anything/issues/2#issuecomment-561210186

This change will make named exports work when used as an ES module, but will require the use of the `--experimental-conditional-exports` flag for commonjs requires in node 13.

### CODE REVIEW

**Comment by mesqueeb** [comment]
@robbtraister is this a work in progress? I'd be glad to merge.

**Comment by robbtraister** [comment]
I marked it as a draft to clarify that this "fix" still requires the experimental flag to work in node 13, and thus that there is a decision to make as to whether to support named exports or naked node 13.

**Comment by robbtraister** [comment]
I can remove draft status if that is what you want to do.

File: build/rollup.js [hunk 1]
```diff
@@ -49,7 +49,7 @@ export default [
   defaults({
     input: 'src/index.ts',
     output: [
-      {file: 'dist/index.cjs.js', format: 'cjs'},
+      {file: 'dist/index.cjs', format: 'cjs', exports: 'named'},
```
Comment (mesqueeb) [inline build/rollup.js]: @robbtraister I don't understand why you have to remove `.js`? 😅

Comment (mesqueeb) [reply build/rollup.js]: I just had this error when I reverted:
> ⠙ (node:44675) Warning: require() of ES modules is not supported.
require() of /Users/lucaban/.ghq/github.com/mesqueeb/merge-anything/dist/index.cjs.js from /Users/lucaban/.ghq/github.com/mesqueeb/merge-anything/test/index.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename index.cjs.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/lucaban/.ghq/github.com/mesqueeb/merge-anything/package.json.

Comment (robbtraister) [reply build/rollup.js]: https://nodejs.org/api/esm.html#esm_package_scope_and_file_extensions

**Comment by mesqueeb** [comment]
@robbtraister ever since 14 February I have changed the library to only have named exports as you can see in the release notes:
https://github.com/mesqueeb/merge-anything/releases/tag/v3.0


Did this solve your issues in node 12+? I believe it did!

**Comment by robbtraister** [comment]
honestly, I haven't experienced this issue in quite a while, and I don't even remember where I ran into it. Thanks for checking. I'll close this.

**Pull request closed without merging**